### ISSUE TITLE
NaNs from exp backwards

### ISSUE THREAD

**Issue by lesterlitch** [pankajmishra000/VT-ADL#10]
Using a different dataset, and consistently getting NaNs returned for the gradient of the line:

log_p = (logvar + a / (torch.exp(logvar))).sum(2) 

from one of the methods of the mdn_loss_function. I see you've used soft_plus on the model outputs, so thinking it shouldn't be because of a negative value for logvar. Did you face this issue before?

**Comment by pankajmishra000** [comment]
yes, I did and you can tweak the function a bit. additionally do check the number of gaussian you are using
